python/sglang/multimodal_gen/runtime/pipelines/diffusers_pipeline.py

```diff
@@ -0,0 +1,490 @@
+            tensors = [T.ToTensor()(img) for img in data]
+            stacked = torch.stack(tensors)
+            if len(tensors) > 1:
+                return stacked.permute(1, 0, 2, 3)  # (T, C, H, W) -> (C, T, H, W)
+            return stacked[0]
+
+        if isinstance(first, torch.Tensor):
+            stacked = torch.stack(data)
+            if len(data) > 1:
+                return stacked.permute(1, 0, 2, 3)
+            return stacked[0]
+
+        if isinstance(first, np.ndarray):
+            tensors = [torch.from_numpy(arr).float() for arr in data]
+            if tensors[0].max() > 1.0:
+                tensors = [t / 255.0 for t in tensors]
+            if tensors[0].ndim == 3:
+                tensors = [t.permute(2, 0, 1) for t in tensors]
+            stacked = torch.stack(tensors)
+            if len(data) > 1:
+                return stacked.permute(1, 0, 2, 3)
+            return stacked[0]
+
+        return None
+
+    def _postprocess_output(self, output: torch.Tensor) -> torch.Tensor:
+        """Post-process output tensor to ensure valid values and correct shape."""
+        output = output.cpu().float()
+
+        # Handle NaN or Inf values
+        if torch.isnan(output).any() or torch.isinf(output).any():
+            logger.warning("Output contains invalid values, fixing...")
+            output = torch.nan_to_num(output, nan=0.5, posinf=1.0, neginf=0.0)
+
+        # Normalize to [0, 1] range if needed
+        min_val, max_val = output.min().item(), output.max().item()
+        if min_val < -0.5 or max_val > 1.5:
+            output = (output + 1) / 2
+
+        output = output.clamp(0, 1)
+
+        # Ensure correct shape for downstream processing
+        output = self._fix_output_shape(output)
+
+        logger.info("Final output tensor shape: %s", output.shape)
+        return output
+
+    def _fix_output_shape(self, output: torch.Tensor) -> torch.Tensor:
+        """Fix tensor shape for downstream processing.
+
+        Expected: (B, C, H, W) for images or (B, C, T, H, W) for videos.
+        """
+        if output.dim() == 5:
+            # Video: (B, T, C, H, W) -> (B, C, T, H, W)
+            return output.permute(0, 2, 1, 3, 4)
+
+        if output.dim() == 4:
+            if output.shape[0] == 1 or output.shape[1] in [1, 3, 4]:
+                return output  # Already (B, C, H, W)
+            # (T, C, H, W) -> (1, C, T, H, W)
+            return output.unsqueeze(0).permute(0, 2, 1, 3, 4)
+
+        if output.dim() == 3:
+            c, h, w = output.shape
+            if c > 4 and w <= 4:
+                output = output.permute(2, 0, 1)
+            if output.shape[0] == 1:
+                output = output.repeat(3, 1, 1)
+            return output.unsqueeze(0)
+
+        if output.dim() == 2:
+            return output.unsqueeze(0).repeat(3, 1, 1).unsqueeze(0)
+
+        return output
+
+    def _build_pipeline_kwargs(self, batch: Req, server_args: ServerArgs) -> dict:
```

Comment:
Question: how do we handle the `__call__` argument mismatches if they occur? For example, if a user specifies `fps` or `num_frames` for an image pipeline, how do we convey it to the user that those arguments will be ignored?